Rocket/Rocket.Core/Commands/RocketCommandManager.cs

```diff
@@ -158,24 +163,60 @@ public void Register(IRocketCommand command, string alias, CommandPriority prior
             foreach(CommandMapping mapping in commandMappings.Instance.CommandMappings.Where(m => m.Class == className && m.Enabled))
             {
                 commands.Add(new RegisteredRocketCommand(mapping.Name.ToLower(), command));
+                commandsDict[mapping.Name] = new RegisteredRocketCommand(mapping.Name.ToLower(), command);
+                if (command.Aliases != null)
+                {
+                    foreach (string Alias in command.Aliases)
+                    {
+                        if (string.IsNullOrEmpty(Alias)) continue;
+                        if(!commandsDict.ContainsKey(Alias))
+                            commandsDict[Alias] = commandsDict[mapping.Name];
+                    }
+                }
                 Logging.Logger.Log("[registered] /" + mapping.Name.ToLower() + " (" + mapping.Class + ")", ConsoleColor.Green);
             }
         }
 
         public void DeregisterFromAssembly(Assembly assembly)
         {
             commands.RemoveAll(rc => getCommandType(rc.Command).Assembly == assembly);
+
+            List<string> cmdsToRemove = new List<string>();
+            foreach (string CMD in commandsDict.Keys)
+            {
+                if (getCommandType(commandsDict[CMD].Command).Assembly == assembly)
+                    cmdsToRemove.Add(CMD);
+            }
+            foreach (string CMD in cmdsToRemove)
+            {
+                commandsDict.Remove(CMD);
+            }
+        }
+        public void DeRegisterCommand(IRocketCommand Command)
+        {
+            commands.RemoveAll(rc => rc.Command == Command);
+
+            List<string> cmdsToRemove = new List<string>();
+            foreach (string CMD in commandsDict.Keys)
+            {
+                if (commandsDict[CMD].Command == Command)
+                    cmdsToRemove.Add(CMD);
+            }
+            foreach (string CMD in cmdsToRemove)
+            {
+                commandsDict.Remove(CMD);
+            }
         }
 
         public double GetCooldown(IRocketPlayer player, IRocketCommand command)
```

Comment:
The cooldown system technically has a memory leak. A cooldown is only removed from the cooldown list when checked.
It is possible to run some kind of timer and clear outdated cooldowns, however it might be best to leave it as it is. The "memory leak" is quite small so it should be fine.

Reply:
If going for the timer method, it would be possible to do something like:
```cs
Task.Run(()=> StartCoroutine(CheckCooldowns));
```
Where CheckCooldowns() loop through the cooldown dictionary and removes all elements that are outdated. This could be ran once per every few minutes or whatever is desired.

Reply:
Looks interesting, I don't think it's a big deal about a memory leak but will see